variables.tf

```diff
@@ -1,9 +1,89 @@
-##############################################################################
-# Input Variables
-##############################################################################
-
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+########################################################################################################################
+# Input variables
+########################################################################################################################
+
+variable "appid_name" {
+  type        = string
+  description = "Name of the AppID resource."
+}
+
+variable "plan" {
+  type        = string
+  description = "Plan for the AppID resource."
+  default     = "graduated-tier"
+  validation {
+    condition     = contains(["graduated-tier", "lite"], var.plan)
+    error_message = "Allowed values for `plan` are \"graduated-tier\", and \"lite\"."
+  }
+}
+
+variable "region" {
+  type        = string
+  description = "Region for the AppID resource."
+}
+
+variable "resource_group_id" {
+  type        = string
+  description = "Resource group ID for the AppID resources."
+}
+
+variable "resource_keys" {
+  description = "The definition of any resource keys to be generated"
+  type = list(object({
+    name                      = string
+    generate_hmac_credentials = optional(bool, false)
+    role                      = optional(string, "Reader")
+    service_id_crn            = string
+  }))
+  default = []
+  validation {
+    # From: https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/resource_key
+    # Writer, Reader, Manager, Administrator, Operator, Viewer, and Editor
+    # Service roles (for Cloud Object Storage) https://cloud.ibm.com/iam/roles
+    # Reader, Writer, Manager, Content Reader, Object Reader, Object Writer
+    # Platform roles (for Cloud Object Storage)
+    # Key Manager
+    condition = alltrue([
+      for key in var.resource_keys : contains(["Writer", "Reader", "Manager", "Administrator", "Operator", "Viewer", "Editor", "Content Reader", "Object Reader", "Object Writer", "Key Manager"], key.role)
```

Comment:
Some of those roles are for COS only. Please update for AppID supported roles only
![image](https://github.com/terraform-ibm-modules/terraform-ibm-appid/assets/29608224/0adf90f9-199e-4c48-9097-a04a2ea90f38)